code/modules/tgui/states/silicon.dm

```diff
@@ -0,0 +1,12 @@
+/**
+ * tgui state: silicon_state
+ *
+ * Checks that the user is an AI/borg.
+ */
+
+var/global/datum/ui_state/tgui_silicon_state/tgui_silicon_state = new /datum/ui_state/tgui_silicon_state
+
+/datum/ui_state/tgui_silicon_state/can_use_topic(src_object, mob/user)
+	if((issilicon(user) && !isghostdrone(user)) || isAIeye(user))
+		return UI_INTERACTIVE
```

Comment:
This gives borgs infinite range which is inconsistent with the other states. The other borg related states require them to be within a certain distance to determine if they can open the UI, if the UI stays open, and if they can interact with the UI. Personally I think we should maintain that behavior here as well

This is also written very differently from the other states which are done by having the datum call the individual procs for each mob type

Reply:
I'd like to warn that removing cyborgs' ability to access doors from a range is a fairly major gameplay change and should probably not be part of a UI overhaul PR.

Reply:
Thinking on this a little more, I've changed my position. Having a check that does simply checks if you are a silicon should be fine. The behavior of allowing borgs access at range can still be achieved by combining with other checks such as `tgui_default_state`

That said this requires silicon state to be broken out into the individual mob procs still to return `UI_CLOSE` when mobs attempt to use it